packages/nimble-tokens/CONTRIBUTING.md

```diff
@@ -61,13 +64,17 @@ For any token metadata changes (e.g. documentation, code snippets, etc.):
    4. Choose to replace any existing files in the `dist/icons/svg` folder.
 
       <img src="docs/ai-export-4.png" width="600"> 
+3. Proceed to the steps below or [create a user story](https://github.com/ni/nimble/issues/new/choose) requesting that the Nimble team perform them. If filing an issue, attach the new and modified SVG files and also `Nimble_Iconography.ai` to the issue.
+
+### Adding icons to Nimble
 
-4. Search for all `<defs>.*</defs>` tags in the exported `.svg` files (in the `packages/nimble-tokens/dist/icons/svg` directory), and remove them. This removes all color from the `.svg` files and allows us to dynamically change the fill color.
+1. In a new branch, copy the Illustrator file to `source/icons/Nimble_Iconography.ai` and the SVG files to `dist/icons/svg`.
+2. Search for all `<defs>.*</defs>` tags in the exported `.svg` files and remove them. This removes all color from the `.svg` files and allows us to dynamically change the fill color.
 
       <img src="docs/find-replace-5.png" width="1000">
 
-5. Confirm the new icon files will build correctly by running: `npm run build -w @ni/nimble-tokens` & `npm run build -w @ni/nimble-components`.
-6. Add metadata for the new icons to `nimble-components\src\icon-base\icon-metadata.ts`.
-7. Generate and build icon components by running `npm run build -w @ni/nimble-components`.
-8. Preview the built files by running: `npm run storybook -w @ni/nimble-components`, and review the **Icons** story to confirm that your changes appear correctly.
-9. Publish a PR with your changes. If there are any new icons, set `changeType` and `dependentChangeType` to minor in the beachball change file.
+3. Confirm the new icon files will build correctly by running: `npm run build -w @ni/nimble-tokens`.
+4. Add metadata for the new icons to `nimble-components\src\icon-base\icon-metadata.ts`.
+5. Generate and build icon components by running `npm run build -w @ni/nimble-components`.
+6. Preview the built files by running: `npm run storybook -w @ni/nimble-components`, and review the **Icons** story to confirm that your changes appear correctly.
```

Comment:
I think part of this step should be ensuring you test the different `severity` attributes of the icons to make sure the color in the svg is removed correctly. I ran into the problem that the color was set on the icon using `style` rather than `defs`, so I hadn't removed it as part of step 2 above.

Reply:
+1 for a visual manual test

but also

> I ran into the problem that the color was set on the icon using style rather than defs, so I hadn't removed it as part of step 2 above

@mollykreis it sounds like your case should have been caught by [this test](https://github.com/ni/nimble/blob/9c85669b7c7d4489a52e91181f8a9522844019d9/packages/nimble-components/src/icon-base/tests/icons.spec.ts#L31). If you happen to remember the details do you think it would have been caught there?

@jattasNI Maybe that test linked above should validate that no defs are included since that seems to be a requirement in these instructions,

Reply:
@rajsite, yes, that test probably would've caught my case. I just happened to catch the issue before I ran any of the tests.

Reply:
Did a manual test and updated the auto test to ensure there are no `defs` children of the SVG. 

I had trouble validating that the autotest worked because whenever I built the icon with a `defs` section added to it, it was stripped out by the time the icon element was generated. I don't understand our generation tooling well enough to know whether that is expected behavior or something wrong with my process. But even if it's not needed, the new assert seems harmless so I decided to keep it. If someone has more info I'm happy to address in a follow up.

Reply:
> I had trouble validating that the autotest worked because whenever I built the icon with a defs section added to it, it was stripped out

The ['preset-default'](https://github.com/ni/nimble/blob/15f605ff76b44a4f7e71e2de88145341791369cb/packages/nimble-tokens/build/svg-to-ts-config.js#L18-L26) enables a bunch of optimization passes and they may be stripping those out. If you were inclined, you could disable those passes by clearing out the plugins, but I'm convinced enough by the test updates as-is 👍